CMakeLists.txt

```diff
@@ -117,6 +123,14 @@ set(COMMON_CXX_FLAGS
   -Wall
 )
 
+# Option for enabling/disabling SIMD flags is for both of debug and release
+if (NTTEC_USE_SIMD)
+  SET(COMMON_CXX_FLAGS "${COMMON_CXX_FLAGS} -msse4.1")
+  add_definitions(-DNTTEC_USE_SIMD)
+else()
+  set(COMMON_CXX_FLAGS "${COMMON_CXX_FLAGS} -mno-sse4.1")
```

Comment:
I think this is not needed. In fact I'm worried that it can disable auto-vectorisation from the compiler.
USE_SIMD should be used to control our explicit use of SIMD, if set we add the required flag, it not set we compile as-is, no?

Reply:
ah yes, you're right. We have flag to exclude simd codes in the case. Thanks

Reply:
I noticed also that -mno-sse leads to a failed compilation due to `llvm/5.0.1/include/c++/v1/unordered_map` and `llvm/5.0.1/include/c++/v1/__hash_table:1161:10`